### ISSUE TITLE
Train and evaluate Decision Tree model

### ISSUE THREAD

**Issue by kgomotsosachane** [The-Elite-Loan-Predictor/loan-predictor#34]
## Objective
Train and evaluate a **Decision Tree model** on the cleaned and preprocessed dataset.

## Tasks
- [ ] **Model Training**: Instantiate and train a Decision Tree Classifier using the training data.
- [ ] **Optimize key parameters (e.g., `max_depth`, `min_samples_split`) to prevent overfitting.
- [ ] **Evaluation & Performance Comparison**: Compare classification reports for **both the Training and Testing datasets** to check for variance and overfitting:
Training Set :  Accuracy during training
 Testing Set :  Accuracy during testing, Precision, Recall, F1-Score, and Confusion Matrix.
